presto-main/src/main/java/com/facebook/presto/operator/aggregation/AccumulatorCompiler.java

```diff
@@ -543,14 +545,32 @@ private static BytecodeBlock generateInputForLoop(
                         .invokeStatic(CompilerOperations.class, "testMask", boolean.class, Block.class, int.class))
                 .ifTrue(loopBody);
 
-        block.append(new ForLoop()
+        BytecodeNode forLoop = new ForLoop()
                 .initialize(new BytecodeBlock().putVariable(positionVariable, 0))
                 .condition(new BytecodeBlock()
                         .getVariable(positionVariable)
                         .getVariable(rowsVariable)
                         .invokeStatic(CompilerOperations.class, "lessThan", boolean.class, int.class, int.class))
                 .update(new BytecodeBlock().incrementVariable(positionVariable, (byte) 1))
-                .body(loopBody));
+                .body(loopBody);
+
+        for (int i = 0; i < parameterVariables.size(); i++) {
+            if (!nullable.get(i)) {
+                Variable variableDefinition = parameterVariables.get(i);
+                forLoop = new IfStatement("if(!(%s instanceof RunLengthEncodedBlock && %s.isNull(0)))", variableDefinition.getName(), variableDefinition.getName())
+                        .condition(and(
+                                variableDefinition.instanceOf(RunLengthEncodedBlock.class),
+                                variableDefinition.invoke("isNull", boolean.class, constantInt(0))))
+                        .ifFalse(forLoop);
+            }
+        }
+
+        block.append(new IfStatement("if(%s > 0)", rowsVariable.getName())
```

Comment:
My understanding is this if-statement is required in case the RLE block has 0 position count, so the `block.isNull(0)`  on line 560 will fail.

Does it make sense to do the empty block check in the line 560 check? i.e. 
```
   if (!(block instanceof RunLengthEncodedBlock && block.getPositionCount() != 0 && block.isNull(0))) {
        // for loop....
   }
```

Reply:
>  this if-statement is required in case the RLE block has 0 position count, so the block.isNull(0) on line 560 will fail.

This is correct. 

> Does it make sense to do the empty block check in the line 560 check? 

I don't have any particular preference, but it seems like empty page check applies even if block not an RLE.